---
title: 'MoTrPAC Webinar: a simple example'
output:
  html_document:
    df_print: paged
editor_options:
  chunk_output_type: inline
---

After downloading the data from the portal, we first load the data into our session.

```{r,warning=F,message=F}
library(data.table)
library(ggplot2)
# load the phenotypic data
load("~/Desktop/all/phenotype/merged_dmaqc_data2019-10-15.RData")
pheno = merged_dmaqc_data
rownames(pheno) = as.character(pheno$vial_label)
# read in RNA-seq data from gastrocnemius
rnaseq_path = "~/Desktop/all/transcriptomics/T68_liver/rna-seq/results/"
rnaseq_d = fread(
  paste(rnaseq_path,"MoTrPAC_rsem_genes_fpkm_liver_v1.txt",sep=""),
  header=T)
print(rnaseq_d[1,1:5])
# read in the protein abundance data
proteomics_path = "~/Desktop/all/proteomics/T68_liver/prot_pr/"
prot_d = fread(
  paste(proteomics_path,"motrpac_t68_prot_pr_ratio_results.txt",sep=""),
  header=T)
print(prot_d[1,1:5])
# look at the rnaseq's tissues
rnaseq_d_vials = colnames(rnaseq_d)[-1]
rnaseq_tissues = pheno[rnaseq_d_vials,
            "specimen.processing.sampletypedescription"]
print(table(rnaseq_tissues))
# sanity check: all prot data vials should be gastrocnemius
prot_d_vials = colnames(prot_d)[-c(1:2)]
print(
  table(pheno[prot_d_vials,
            "specimen.processing.sampletypedescription"])
)
```

We now match the samples from the two datasets. 
We use the "pid", which is the animal id.

```{r,warning=F,message=F}
# Get pid-based matched data tables
# 1. rnaseq data: map to pid
rnaseq_d_pid = as.character(pheno[rnaseq_d_vials,"pid"])
rnaseq_liver_d = data.frame(rnaseq_d[,..rnaseq_d_vials])
rownames(rnaseq_liver_d) = rnaseq_d$gene_id
colnames(rnaseq_liver_d) = rnaseq_d_pid
# 2. proteomics: map to pid
prot_d_pid = as.character(pheno[prot_d_vials,"pid"])
prot_liver_d = data.frame(prot_d[,..prot_d_vials])
rownames(prot_liver_d) = prot_d$protein_id
colnames(prot_liver_d) = prot_d_pid
# 3. shared pids
# keep a mapping from pid to vial_label
# useful for getting phenotypes later
pid2vial = rnaseq_d_vials;names(pid2vial) = rnaseq_d_pid
# intersect and match the datasets
shared_pids = intersect(prot_d_pid,rnaseq_d_pid)
rnaseq_liver_d = log2(rnaseq_liver_d[,shared_pids]+1) # log2 (+1 to avoid inf/negative)
prot_liver_d = prot_liver_d[,shared_pids] # keep the log ratios as is
```

Load tools for mapping molecule ids. These are not a must for the code below, but are shown as a useful reference.

```{r,warning=F,message=F}
library("org.Rn.eg.db")
ensembl2entrez = as.list(org.Rn.egENSEMBL2EG)
entrez2ensembl = as.list(org.Rn.egENSEMBL)
refseq2entrez = as.list(org.Rn.egREFSEQ2EG)
entrez2symbol  = as.list(org.Rn.egSYMBOL)
```

We can now look at a signle gene: *Bdh1*.

```{r}
ensembl_id = "ENSRNOG00000001736"
prot_id = "NP_446447.2"
x1 = as.numeric(rnaseq_liver_d[ensembl_id,] )
x2 = as.numeric(prot_liver_d[prot_id,] )
tp = pheno[pid2vial[colnames(rnaseq_liver_d)],"animal.key.anirandgroup"]
# keep the time points ordered using the levels argument
tp = factor(tp,levels = c(
  "Control - IPE","Exercise - IPE","Exercise - 1 hr",
  "Exercise - 4 hr","Exercise - 24 hr"
))
# define a single data frame, not a must, but easy to work with
df = data.frame(
  rnaseq = x1, protein = x2, group = tp
)
print(cor(df$rnaseq,df$protein,method="spearman"))

# A simple 2d plot
ggplot(df, aes(x=rnaseq, y=protein, color=tp)) + geom_point()

# Boxplot by the study arms
par(mar=c(7,5,1,1),mfrow = c(1,2))
boxplot(rnaseq~tp,data=df,las=2,xlab="", ylab = "log2 FPKM")
boxplot(protein~tp,data=df,las=2,xlab = "", ylab = "log2 ratio vs. ref")

```

```{r,warning=F,message=F,eval=F}
# # Not run
# for (prot_id in rownames(prot_liver_d)){
#   prot_id_cleaned = gsub("\\.\\d","",prot_id)
#   ent_id = refseq2entrez[[prot_id_cleaned]]
#   if(is.null(ent_id) || is.na(ent_id)){next}
#   ensembl_id = entrez2ensembl[[ent_id]]
#   if(is.null(ensembl_id) || is.na(ensembl_id)){next}
#   gene_symbol  = entrez2symbol[[ent_id]]
#   x1 = rnaseq_liver_d[ensembl_id[1],] # take first index
#   x2 = prot_liver_d[prot_id[1],] # take first index
#   non_nas = !is.na(x1) & !is.na(x2)
#   if(sd(x1[non_nas])==0 || sd(x2[non_nas])==0){next}
#   rho = cor(x1[non_nas],x2[non_nas],method="spearman")
#   if(rho > 0.5){
#     print(paste(prot_id,ent_id,ensembl_id,gene_symbol),sep=",")
#     print(rho)
#   }
# }
```
